packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -212,22 +213,17 @@ impl Abigen {
     // It's expected to come in as `"struct T"` or `"enum T"`.
     // `T` is a native `high-level language` or Rust type if it matches exactly one of
     // the reserved strings, such as "Address", "ContractId", "Option" or "Result"
-    pub fn is_native_type(type_field: &str) -> bool {
-        const CONTRACT_ID_NATIVE_TYPE: &str = "ContractId";
-        const ADDRESS_NATIVE_TYPE: &str = "Address";
-        const OPTION_NATIVE_TYPE: &str = "Option";
-        const RESULT_NATIVE_TYPE: &str = "Result";
-
-        let split: Vec<&str> = type_field.split_whitespace().collect();
-
-        if split.len() > 2 {
-            return false;
-        }
-
-        split[1] == CONTRACT_ID_NATIVE_TYPE
-            || split[1] == ADDRESS_NATIVE_TYPE
-            || split[1] == OPTION_NATIVE_TYPE
-            || split[1] == RESULT_NATIVE_TYPE
+    pub fn is_native_type(type_field: &str) -> anyhow::Result<bool> {
+        let name = extract_custom_type_name_from_abi_type_field(type_field)?;
+
+        // "RawVec" is part of the Vec structure. Not used in the SDK and thus
+        // not generated.
+        Ok(
+            ["ContractId", "Address", "Option", "Result", "Vec", "RawVec"]
```

Comment:
These should still be `const`s extracted away from this function, though. Easier to maintain/find/refer to. Plus, there's a new `"Identity"` one in master; you'll see it once you sync with it!

Reply:
Hopefully, I can sway you with the following :D 

> These should still be consts 

Godbolt shows both `let a = "something"` and `const a: &str = "something"` results in the same asm.


> extracted away from this function
They have meaning only in this function. I'd rather not place them somewhere else.

> Easier to maintain/find/refer to.
They are inside a 10line function. Only to be used here. The keywords are used only once and in such a way that their usage gives them meaning -- i.e. `if my type matches any of the following ones`. 

> there's a new "Identity" one in master; you'll see it once you sync with it!
I've added it just now :)

Reply:
> Godbolt shows both let a = "something" and const a: &str = "something" results in the same asm.

My suggestion is _really_ not related to the generated asm 😂. 100% related to readability. But this: 

> They have meaning only in this function. I'd rather not place them somewhere else.

Makes it ok for it to be inside this function (and, thus, not a `const`). The moment we start referencing this from other places, then we should extract it out. But yeah, not needed right now!